packages/next/src/build/entries.ts

```diff
@@ -148,45 +150,32 @@ export async function getStaticInfoIncludingLayouts({
       dir = join(dir, '..')
     }
 
+    // Reverse the layout files so we can use unshift to add them to the
+    // segments array.
+    layoutFiles.reverse()
```

Comment:
this part is a little strange to me, could you clarify in the comment why we do this?